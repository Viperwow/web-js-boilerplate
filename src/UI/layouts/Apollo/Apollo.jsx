import React from 'react';
import {ApolloProvider} from 'react-apollo';
import {
  getApolloClient,
  initApolloClient,
} from 'src/helpers/apollo';
import App from 'src/UI/layouts/App';

const Apollo = () => {
  initApolloClient();

  return (
    <ApolloProvider client={getApolloClient()}>
      <App />
    </ApolloProvider>
  );
};

export default Apollo;
