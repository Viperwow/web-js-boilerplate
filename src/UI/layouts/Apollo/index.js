export {default} from 'src/UI/layouts/Apollo/Apollo';
