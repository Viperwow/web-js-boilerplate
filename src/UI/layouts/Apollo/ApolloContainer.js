import prepare from 'src/common/prepare';
import withOnError from 'src/common/withOnError';
import {initApolloClient} from 'src/helpers/apollo';
import {compose} from 'src/helpers/utility';
import Apollo from './Apollo';

const initialPreparation = () => {
  initApolloClient();
};

export default compose(
  prepare({preparation: initialPreparation}),
  withOnError,
)(Apollo);
