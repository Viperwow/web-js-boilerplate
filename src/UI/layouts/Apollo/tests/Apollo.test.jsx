import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Apollo from '../Apollo';

jest.unmock('../Apollo');

describe('Apollo component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<Apollo />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
