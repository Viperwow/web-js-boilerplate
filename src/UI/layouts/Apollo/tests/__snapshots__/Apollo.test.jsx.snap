// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apollo component should render correctly With default props 1`] = `
<ApolloProvider
  client={Object {}}
>
  <App />
</ApolloProvider>
`;
